From f79179306ab5a3e64b23e70de2b30d15ca027194 Mon Sep 17 00:00:00 2001
From: Min RK <benjaminrk@gmail.com>
Date: Tue, 19 Nov 2019 10:07:27 +0100
Subject: [PATCH] set install_name to $(SO_MAIN) on mac

install_name is the mac equivalent of soname

ensures runtime compatibility of dependents compiled against the same SO version
---
 GraphBLAS/Tcov/Makefile                  | 2 +-
 GraphBLAS/alternative/Makefile           | 2 +-
 SuiteSparse_config/SuiteSparse_config.mk | 1 +
 3 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/GraphBLAS/Tcov/Makefile b/GraphBLAS/Tcov/Makefile
index 1a1c6494..f7c184d2 100644
--- a/GraphBLAS/Tcov/Makefile
+++ b/GraphBLAS/Tcov/Makefile
@@ -35,7 +35,7 @@ ifeq ($(UNAME),Darwin)
     # Mac
     CFLAGS += -fno-common
     SO_NAME = libgraphblas_tcov.dylib
-    SO_OPTS += -dynamiclib -shared -undefined dynamic_lookup
+    SO_OPTS += -dynamiclib -shared -Wl,-install_name -Wl,$(SO_NAME) -undefined dynamic_lookup
 else
     # Linux
     SO_NAME = libgraphblas_tcov.so
diff --git a/GraphBLAS/alternative/Makefile b/GraphBLAS/alternative/Makefile
index 703758b9..c67dc9cf 100644
--- a/GraphBLAS/alternative/Makefile
+++ b/GraphBLAS/alternative/Makefile
@@ -31,7 +31,7 @@ ifeq ($(UNAME),Darwin)
     # Mac
     CFLAGS += -fno-common
     SO_NAME = libgraphblas.dylib
-    SO_OPTS += -dynamiclib -shared -undefined dynamic_lookup
+    SO_OPTS += -dynamiclib -shared  -Wl,-install_name -Wl,$(SO_NAME) -undefined dynamic_lookup
 else
     # Linux
     SO_NAME = libgraphblas.so
diff --git a/SuiteSparse_config/SuiteSparse_config.mk b/SuiteSparse_config/SuiteSparse_config.mk
index 64c9bbd5..b9ca7921 100644
--- a/SuiteSparse_config/SuiteSparse_config.mk
+++ b/SuiteSparse_config/SuiteSparse_config.mk
@@ -464,6 +464,7 @@ else
         SO_TARGET = $(LIBRARY).$(VERSION).dylib
         SO_OPTS  += -dynamiclib -compatibility_version $(SO_VERSION) \
                     -current_version $(VERSION) \
+                    -Wl,-install_name -Wl,$(SO_MAIN) \
                     -shared -undefined dynamic_lookup
         # When a Mac *.dylib file is moved, this command is required
         # to change its internal name to match its location in the filesystem:
-- 
2.23.0

